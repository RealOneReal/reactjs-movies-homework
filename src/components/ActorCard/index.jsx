import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import { truncateString } from "../../utils/stringUtils";

const ActorCard = ({ id, profile_path, name, character }) => {
  const handleClick = () => {
    console.log(id);
  };

  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const styles = {
    card: {
      width: "150px",
      height: "225px",
      cursor: "pointer",
      ":hover ": {
        transform: "translateY(-16px)",
        transition: ".3s ease",
      },
      bgcolor: "secondary.dark",
    },
    cardMedia: { height: "170px", borderRadius: "10px" },
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia component="img" image={profile_path} sx={styles.cardMedia} />
      <CardContent sx={{ p: "4px" }}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
