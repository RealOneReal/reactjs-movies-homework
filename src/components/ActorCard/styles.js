const styles = {
  card: {
    width: "150px",
    height: "225px",
    cursor: "pointer",
    ":hover ": {
      transform: "translateY(-16px)",
      transition: ".3s ease",
    },
    bgcolor: "secondary.dark",
  },
  cardMedia: { height: "170px", borderRadius: "10px" },
  cardContent: { p: "4px" },
};

export default styles;
