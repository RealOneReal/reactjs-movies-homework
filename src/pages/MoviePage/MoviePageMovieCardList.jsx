import { Grid } from "@mui/material";
import MovieCard from "../../components/MovieCard/MovieCard";
import DataFromTMDB from "../../mock-data/mockMovies";

const MoviesPageMovieCardList = () => {
  // todo: get recommendations from Redux ,it`s temporary implementation

  const results = DataFromTMDB?.results.slice(0, 5);
  return (
    <Grid container spacing={1} justifyContent="center">
      {results &&
        results.map((card) => (
          <Grid item key={card.id}>
            <MovieCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};

export default MoviesPageMovieCardList;
