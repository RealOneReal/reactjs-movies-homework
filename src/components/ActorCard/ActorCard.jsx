import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import styles from "./styles";
import { truncateString } from "../../utils/stringUtils";
import Fallback from "../../assets/broken-image.png";

const ActorCard = ({ id, profile_path, name, character }) => {
  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const handleClick = () => {
    console.log(id);
  };

  const handleMediaFallback = (event) => {
    event.target.style.objectFit = "contain";
    return (event.target.src = Fallback);
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia
        component="img"
        alt={actorName}
        src={profile_path}
        sx={styles.cardMedia}
        onError={handleMediaFallback}
      />
      <CardContent sx={styles.cardContent}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
