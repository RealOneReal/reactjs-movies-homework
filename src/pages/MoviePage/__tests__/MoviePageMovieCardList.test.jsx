import { render } from "@testing-library/react";
import MoviePageMovieCardList from "../MoviePageMovieCardList";
import ActorCard from "../../../components/ActorCard/ActorCard";
import DataFromTMDB from "../../../mock-data/mockMovies";

jest.mock("../../../mock-data/mockMovies");
jest.mock("../../../components/ActorCard/ActorCard", () => () => "ActorCard");

describe("MoviePageMovieCardList", () => {
  it("should be rendered", () => {
    const { container } = render(<MoviePageMovieCardList />);
    expect(container).toMatchSnapshot();
  });
});
