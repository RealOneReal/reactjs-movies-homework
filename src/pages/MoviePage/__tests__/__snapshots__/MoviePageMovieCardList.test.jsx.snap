// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageMovieCardList should be rendered 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
  />
</div>
`;
